### PR TITLE
Fix ability to inherit non subnet aware locations from environment

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/SubmittedEnvironmentDeserializer.kt [hunk 1]
```diff
@@ -27,7 +30,7 @@ class SubmittedEnvironmentDeserializer : StdNodeBasedDeserializer<SubmittedEnvir
       val notifications: Set<NotificationConfig> = convert(root, "notifications") ?: emptySet()
       val locations: SubnetAwareLocations? = convert(root, "locations")
       val resources: Set<SubmittedResource<*>> = copy().run {
-        injectableValues = InjectableValues.Std(mapOf("locations" to locations))
+        injectableValues = InjectableLocations(locations)
```
Comment (robfletcher) [inline keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/SubmittedEnvironmentDeserializer.kt]: Fixed by creating my own implementation of this `InjectableValues` thing that can detect the desired type

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/SubmittedEnvironmentDeserializer.kt [hunk 2]
```diff
@@ -44,3 +47,21 @@ class SubmittedEnvironmentDeserializer : StdNodeBasedDeserializer<SubmittedEnvir
...
+    valueId: Any,
+    context: DeserializationContext,
+    forProperty: BeanProperty,
+    beanInstance: Any?
+  ): Any? {
+    val value = super.findInjectableValue(valueId, context, forProperty, beanInstance) as? SubnetAwareLocations
+    return when {
+      value == null -> null
+      forProperty.type.isTypeOrSubTypeOf(SimpleLocations::class.java) -> value.toSimpleLocations()
+      else -> value
```
Comment (robfletcher) [inline keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/SubmittedEnvironmentDeserializer.kt]: Converts the `locations` value if necessary

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/SubmittedEnvironmentDeserializer.kt (hunk 1 above)
Comment (asher) [reply keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/SubmittedEnvironmentDeserializer.kt]: Do we lose any flexibility here for injecting additional (non-location) values in the same way?

Comment (robfletcher) [reply keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/SubmittedEnvironmentDeserializer.kt]: Yeah, we'd have to amend this as & when we do. If that's the case we may want an extension that allows for an optional conversion based on context.

**Pull request merged**